### PR TITLE
add option proxy_headers for k8s modules

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

Fix on kubernetes-python module https://github.com/kubernetes-client/python/pull/1128
#55 

##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

k8s authentication
##### ADDITIONAL INFORMATION

```paste below

```
We can now authenticate on k8s cluster using a proxy with proxy headers

```
- hosts: localhost
  tasks:
  - name: create namespace
    kubernetes.core.k8s:
      defintion:
         name: test
         kind: namespace
      proxy: "http://localhost:3001/"
      proxy_headers: 
          proxy_basic_auth: "username:password"

```

### CODE REVIEW

File: plugins/module_utils/common.py [hunk 1]
```diff
@@ -173,6 +172,9 @@ def auth_set(*names):
         if key in AUTH_ARG_MAP.keys() and value is not None:
             if key == 'api_key':
                 setattr(configuration, key, {'authorization': "Bearer {0}".format(value)})
+            elif key == 'proxy_headers':
+                headers = urllib3.make_headers(proxy_basic_auth=value)
```
Comment (Akasurde) [inline plugins/module_utils/common.py:176]: Do you think we can have other `proxy_headers` like `user-agent`?

**Review by Akasurde — CHANGES_REQUESTED** [review]
as per above

File: plugins/module_utils/common.py (hunk 1 above)
Comment (abikouo) [reply plugins/module_utils/common.py:176]: @Akasurde could you please clarify ?

Comment (Akasurde) [reply plugins/module_utils/common.py:176]: I think `proxy_headers` can be a `dict` instead of `str` like

```
    proxy_headers:
       proxy_basic_auth: 'username:password'
       user_agent: 'Batman'
```

Something similar to 
https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html?highlight=proxy_headers#urllib3.util.make_headers

File: plugins/doc_fragments/k8s_auth_options.py [hunk 2]
```diff
@@ -77,9 +77,26 @@ class ModuleDocFragment(object):
     type: str
   proxy_headers:
     description:
-    - The login for the HTTP proxy. Can also be specified via K8S_AUTH_PROXY_HEADERS environment variable.
-    type: str
+    - The login for the HTTP proxy.
+    - Documentation can be found here 'https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html?highlight=proxy_headers#urllib3.util.make_headers'
```
Comment (Akasurde) [inline plugins/doc_fragments/k8s_auth_options.py:81]:
```suggestion
    - Documentation can be found here U(https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html?highlight=proxy_headers#urllib3.util.make_headers).
```

File: plugins/doc_fragments/k8s_auth_options.py [hunk 3]
```diff
@@ -77,9 +77,26 @@ class ModuleDocFragment(object):
     type: str
   proxy_headers:
     description:
-    - The login for the HTTP proxy. Can also be specified via K8S_AUTH_PROXY_HEADERS environment variable.
-    type: str
+    - The login for the HTTP proxy.
```
Comment (Akasurde) [inline plugins/doc_fragments/k8s_auth_options.py:80]:
```suggestion
    - The Header used for the HTTP proxy.
```

File: molecule/default/molecule.yml [hunk 4]
```diff
@@ -12,6 +12,16 @@ lint: |
   flake8
 platforms:
   - name: instance-kind
+    image: sameersbn/squid:3.5.27-2
```
Comment (Akasurde) [inline molecule/default/molecule.yml:15]: Is this change related?

File: .gitignore [hunk 5]
```diff
@@ -12,3 +12,4 @@ changelogs/.plugin-cache.yaml
 # Temporary test files.
 tests/output
 tests/integration/cloud-config-*
+.cache
```
Comment (Akasurde) [inline .gitignore:15]: I would open a separate PR to modify `.gitignore` file.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: molecule/default/molecule.yml (hunk 4 above)
Comment (abikouo) [reply molecule/default/molecule.yml:15]: testing, will be removed

**Review by Akasurde — APPROVED** [review]
LGTM

**Review by gravesm — APPROVED** [review]
